superset/common/query_context.py

```diff
@@ -0,0 +1,24 @@
+# pylint: disable=R
+from typing import Dict
+
+from superset import db
+from superset.connectors.connector_registry import ConnectorRegistry
+from .query_object import QueryObject
+
+
+class QueryContext:
+    """
+    The query context contains the query object and additional fields necessary
+    to retrieve the data payload for a given viz.
+    """
+    # TODO: Type datasource and query_object dictionary with TypedDict when it becomes
+    # a vanilla python type https://github.com/python/mypy/issues/5288
+    def __init__(
+            self,
+            datasource: Dict,
+            query_object: Dict,
+    ):
+        self.datasource = ConnectorRegistry.get_datasource(datasource.get('type'),
+                                                           datasource.get('id'),
+                                                           db.session)
+        self.query_object = QueryObject(**query_object)
```

Comment:
As we discussed, the query_object should be an array, or should we make it as an array of `QueryObject`?

Reply:
Yep, I can change this to an array of `QueryObject`s